### PR TITLE
Require running on Java 11

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/run-presto.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ set -x
 
 tar xf /docker/presto-server.tar.gz -C /docker
 exec /docker/presto-server-*/bin/launcher \
+  -Dresto-temporarily-allow-java8=true \
```
Comment (kokosing) [inline presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/run-presto.sh]: Typo

File: presto-main/src/main/java/io/prestosql/server/PrestoSystemRequirements.java [hunk 2]
```diff
@@ -101,6 +101,14 @@ private static void verifyJavaVersion()
             return;
         }
 
+        if (!Boolean.getBoolean("presto-temporarily-allow-java8")) {
+            failRequirement("" +
+                    "Future versions of Presto will require Java 11 starting in April 2020.\n\n" +
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/server/PrestoSystemRequirements.java]: I think it should be like 6 months from now.

Comment (findepi) [reply presto-main/src/main/java/io/prestosql/server/PrestoSystemRequirements.java]: This date is the one announced previously on Slack. https://prestosql.slack.com/archives/CFLB9AMBN/p1580870213234300

However, the code build today will continue to work and we will unlikely do a release on April 1st, so I'd use a different wording here

```suggestion
                    "Future versions of Presto will require Java 11 after March 2020.\n\n" +
```

File: presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/run-presto.sh (hunk 1 above)
Comment (findepi) [reply presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/run-presto.sh]:
```suggestion
  -Dpresto-temporarily-allow-java8=true \
```

File: presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java [hunk 3]
```diff
@@ -63,6 +63,8 @@ private static void testServer(String baseImage, String rpmHostPath, String expe
         String command = "" +
                 // install RPM
                 "yum localinstall -q -y " + rpm + "\n" +
+                // temporary hack to allow Java 8
+                "echo '-Dpresto-temporarily-allow-java8=true' >> /etc/jvm.config\n" +
```
Comment (findepi) [inline presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java]: This should be applied only when running on `centos7-oj8` image

File: presto-main/pom.xml [hunk 4]
```diff
@@ -377,6 +377,20 @@
...
 
     <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <configuration>
+                        <systemProperties>
+                            <presto-temporarily-allow-java8>true</presto-temporarily-allow-java8>
```
Comment (findepi) [inline presto-main/pom.xml]: Add an explanatory comment

File: presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java (hunk 3 above)
Comment (electrum) [reply presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java]: There's no harm in having it for both -- it's ignored on Java 11. Do you have a specific concern with having it for both?

Comment (findepi) [reply presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java]: I thought that we could verify this property is _not_ needed when running on 11.

Comment (electrum) [reply presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java]: Make sense. I'll update it.

**Pull request merged**